### PR TITLE
[O2B-762] updatable detectors

### PR DESCRIPTION
**Pull request by JikHuijberts**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for developers:
- A new endpoint `api/runs/:runId/detector/:detectorId` that has a `quality` body that takes `good, bad, none` values to update. 

Changes made to the database:
-

### CODE REVIEW

File: lib/server/services/runDetector/getRunDetector.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const { utilities: { TransactionHelper } } = require('../../../database');
+
+/**
+ * Find and return an run detector model by its id
+ *
+ * @param {string} runNumber the runNumber of the run to find
+ * @param {string} detectorId the id of the detector to find
+ * @param {function|null} qbConfiguration function called with the run detector find query builder as parameter to add specific configuration to
+ *     the query
+ * @param {Object|null} [transaction] optionally the transaction in which one the log creation is executed
```
Comment (martinboulais) [inline lib/server/services/runDetector/getRunDetector.js:25]: It may be wrong in my own code, but you can specify the default value of optional code in the square brackets `@param {function|null} [qbConfiguration=null]`

File: lib/server/services/runDetector/getRunDetectorOrFail.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+const { getRunDetector } = require('./getRunDetector.js');
+const { NotFoundError } = require('../../errors/NotFoundError.js');
+const { utilities: { TransactionHelper } } = require('../../../database');
+
+/**
+ * Find an run detector model by its id and reject with a NotFoundError if none is found
```
Comment (martinboulais) [inline lib/server/services/runDetector/getRunDetectorOrFail.js:19]: It is not by its id but by the related run and detector

File: lib/server/services/runDetector/getRunDetectorOrFail.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * @param {number} runNumber the id of the run detector to find
```
Comment (martinboulais) [inline lib/server/services/runDetector/getRunDetectorOrFail.js:21]: runNumber the id ? Maybe as you put it before `runNumber the identifier`

File: test/e2e/runDetectors.test.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                    }
+                    // Response must satisfy the OpenAPI specification
+                    expect(res).to.satisfyApiSpec;
+                    expect(res.body.errors[0].detail)
+                        .to
+                        .equal('This run\'s detector with runNumber: (106) and with detector Id: (9999999) could not be found');
+                    done();
+                });
+        });
+        it('should return 200 when the right quality is given', (done) => {
```
Comment (martinboulais) [inline test/e2e/runDetectors.test.js:46]: This text is not related to the test

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/server/services/runDetector/getRunDetector.js (hunk 1 above)
Comment (JikHuijberts) [reply lib/server/services/runDetector/getRunDetector.js]: done

File: lib/server/services/runDetector/getRunDetectorOrFail.js (hunk 2 above)
Comment (JikHuijberts) [reply lib/server/services/runDetector/getRunDetectorOrFail.js]: done

File: lib/server/services/runDetector/getRunDetectorOrFail.js (hunk 3 above)
Comment (JikHuijberts) [reply lib/server/services/runDetector/getRunDetectorOrFail.js]: done

File: package.json [hunk 5]
```diff
@@ -12,7 +12,7 @@
     "start:prod": "node lib/main.js",
     "test": "mocha --exit --timeout 0",
     "docker-run": "docker-compose -f docker-compose.yml -f docker-compose.dev.yml up --build",
-    "docker-test": "docker-compose -p test -f docker-compose.yml -p test -f docker-compose.test.yml up --build --abort-on-container-exit"
+    "docker-test": "docker-compose -f docker-compose.yml -f docker-compose.test.yml up --build --abort-on-container-exit"
```
[the remark this answers is not in the document]
Comment (JikHuijberts) [reply package.json]: reversed

File: test/e2e/runDetectors.test.js (hunk 4 above)
Comment (JikHuijberts) [reply test/e2e/runDetectors.test.js]: fixed

File: test/e2e/runDetectors.test.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+const chaiResponseValidator = require('chai-openapi-response-validator');
+const { server } = require('../../lib/application');
+
+const { expect } = chai;
+
+chai.use(chaiResponseValidator(path.resolve(__dirname, '..', '..', 'spec', 'openapi.yaml')));
+
+module.exports = () => {
+    describe('PATCH /api/runs/:runNumber/detector/:detectorId ', () => {
+        it('should return 400 when detectorId is wrong', (done) => {
```
[the remark this answers is not in the document]
Comment (JikHuijberts) [reply test/e2e/runDetectors.test.js]: done

File: lib/domain/dtos/UpdateRunDetectorDto.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+const EntityIdDto = require('./EntityIdDto');
+
+const BodyDto = Joi.object({
+    quality: Joi.string().valid(RunDetectorQuality.Bad, RunDetectorQuality.Good).required(),
+});
+const QueryDto = Joi.object({
+    token: Joi.string(),
+});
+
+const UpdateTagDto = Joi.object({
```
Comment (martinboulais) [inline lib/domain/dtos/UpdateRunDetectorDto.js:25]: Please rename this

**Review by martinboulais — DISMISSED** [review]

**Review by martinboulais — APPROVED** [review]
